network/network.go

```diff
@@ -0,0 +1,122 @@
+package network
+
+import (
+	"context"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/network"
+	"github.com/google/uuid"
+
+	"github.com/testcontainers/testcontainers-go"
+)
+
+// New creates a new network with a random UUID name, calling the already existing GenericNetwork APIs.
+// Those existing APIs are deprecated and will be removed in the future, so this function will
+// implement the new network APIs when they will be available.
+// By default, the network is created with the following options:
+// - Driver: bridge
+// - Attachable: true
+// - Labels: the Testcontainers for Go generic labels, to be managed by Ryuk. Please see the GenericLabels() function
+// And those options can be modified by the user, using the CreateModifier function field.
+func New(ctx context.Context, opts ...NetworkCustomizer) (*testcontainers.DockerNetwork, error) {
+	nc := types.NetworkCreate{
+		Driver:     "bridge",
+		Attachable: true,
+		Labels:     testcontainers.GenericLabels(),
+	}
+
+	for _, opt := range opts {
+		opt.Customize(&nc)
+	}
+
+	//nolint:staticcheck
+	netReq := testcontainers.NetworkRequest{
+		Driver:         nc.Driver,
+		CheckDuplicate: nc.CheckDuplicate,
+		Internal:       nc.Internal,
+		EnableIPv6:     nc.EnableIPv6,
+		Name:           uuid.NewString(),
+		Labels:         nc.Labels,
+		Attachable:     nc.Attachable,
+		IPAM:           nc.IPAM,
```

Comment:
Looking at the networks configuration, `checkDuplicate` looks like something we can use in java and make it default in both. But, wonder if I missed something from the first implementation. The rest of them offer much flexibility when creating networks and, at least, in Java there is no request to add all those configurations.

Reply:
The previous impl provided a way to modify any of those values, that's why I added a functional option for each of them.

Reply:
My suggestion is to start supporting the same as in Java and then see if support for the rest of them are needed. Maybe a networkModifier would be used for the rest of them.

Reply:
The thing is that in Go we already supported all those fields (there are more in the Docker type). That's why I kept them